data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/CloudEventOverridesMutator.java

```diff
@@ -42,6 +46,21 @@ public CloudEvent apply(CloudEvent cloudEvent) {
         return builder.build();
     }
 
+    private CloudEvent maybeDeserializeFromHeaders(ConsumerRecord<Object, CloudEvent> record) {
+        if (record.value() != null) {
+            return record.value();
+        }
+        // A valid CloudEvent in the CE binary protocol binding of Kafka
+        // might be composed by only Headers.
+        //
+        // KafkaConsumer doesn't call the deserializer if the value
+        // is null.
+        //
+        // That means that we get a record with a null value and some CE
+        // headers even though the record is a valid CloudEvent.
+        return cloudEventDeserializer.deserialize(record.topic(), record.headers(), null);
+    }
```

Comment:
This logic is duplicated from `RecordDispatcherImpl`, can we extract a common method, usable by both `CloudEventOverridesMutator` and `RecordDispatcherImpl`